Use tox for testing/development

nicoddemus opened the pull request:
Tox is the de-facto standard tool for testing in Python today. It automatically:

1. Creates a source dist of the package;
2. Creates a new virtual environment for each Python version you want to support;
3. Runs the tests using the code installed on the package. This is important to catch packaging mistakes like forgetting to include a data file or python package inside your source dist.

By using tox you reduce the entry barrier for developers to contribute. Users which know `tox` know that they just have to type `tox` at the root of your project to be able to contribute.

Basically every serious project in Python today uses `tox` for testing automation. Just a few examples: [pip](https://github.com/pypa/pip), [pipfile](https://github.com/pypa/pipfile), [virtualenv](https://github.com/pypa/virtualenv), [requests](https://github.com/requests/requests), etc. Is actually hard to find a project which doesn't use `tox` I think. 😁 

Well this is just a suggestion. I understand perfectly if you guys don't want to add this right now and want to reject the PR, no problem. 👍

rougeth commented:
@nicoddemus I was about to open an issue to config tox! Awesome you already did it!

Did you saw that we started the documentation? \o/ Could you move what you wrote on `README.md` to a `contribution.rst` page?

nicoddemus commented:
On this PR? Sure, will do!

rougeth commented:
Yep, but just move it there, no need to fill more info, I'll spend some time on docs this weekend.

nicoddemus commented:
Are you planning on hosting the docs on readthedocs I assume? Did you already reserve the name of the project?

rougeth commented:
It's already there, http://bottery.readthedocs.io/en/latest/. But I want to hold it a bit more.

nicoddemus commented:
Please go to the AppVeyor settings page and check the setting "Skip branches without appveyor.yml". It is in the first page. Well it won't matter anyway once we have appveyor running anyway, but it is the right thing to do. 😁

nicoddemus commented:
I updated the PR with the contribution page and added a tox environment which can be used to build the docs locally and also builds the docs on Travis, just to ensure they are correct (without missing documents, orphan pages, etc).

rougeth commented on CONTRIBUTING.rst:
```diff
@@ -0,0 +1,35 @@
+
+Use PEP-8 for code style and `isort <https://pypi.python.org/pypi/isort>`_ to sort your imports.
+
+Bottle uses `tox <http://tox.readthedocs.io>`_ for testing and general development.
```
*Battery :)

nicoddemus replied:
Oops, fixed

rougeth commented on tox.ini:
```diff
@@ -0,0 +1,28 @@
+changedir = docs
+deps =
+    sphinx
+    PyYAML
```
Why we need `PyYAML`?

nicoddemus replied:
Indeed we don't, thanks

rougeth commented:
Thank you veeery much @nicoddemus!

nicoddemus commented:
Nice, thanks!
